### PR TITLE
How to customize a State Machine?

### PR DESCRIPTION
**Pull request by AdamKasp**
[Docs - Customizing State Machines](https://docs.sylius.com/en/1.4/customization/state_machine.html)

### CODE REVIEW

File: config/packages/sylius_mailer.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+sylius_mailer:
+    sender:
+        name: review confirmation
```
Comment (GSadee) [inline config/packages/sylius_mailer.yaml]: **R**eview

File: config/packages/sylius_mailer.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+        address: example@example.com
+    emails:
+        review_confirmation:
+            subject: A new review has been added
+            template: Emails/reviewConfirmationNotification.html.twig
```
Comment (GSadee) [inline config/packages/sylius_mailer.yaml]: Simply `reviewConfirmation.html.twig`?

File: features/receiving_confirmation_email_after_review_added.feature [hunk 3]
```diff
@@ -0,0 +1,19 @@
+@managing_product_reviews
+Feature: Receiving confirmation email after adding review
```
Comment (GSadee) [inline features/receiving_confirmation_email_after_review_added.feature]: Change the filename

File: features/receiving_confirmation_email_after_review_added.feature [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    In order to send email with confirmation after review accepted
+    As a Administrator
```
Comment (GSadee) [inline features/receiving_confirmation_email_after_review_added.feature]: `an`

File: src/ProductReview/Mailer/ConfirmationMailer.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    private $emailSender;
+
+    public function __construct(SenderInterface $emailSender)
+    {
+        $this->emailSender = $emailSender;
+    }
+
+    public function sendEmail(ReviewInterface $review): void
+    {
+        $this->emailSender->send("review_confirmation",
```
Comment (GSadee) [inline src/ProductReview/Mailer/ConfirmationMailer.php]: `"` -> `'`

File: src/ProductReview/Mailer/ConfirmationMailer.php [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+            [$review->getAuthor()->getEmail()],
+            ['productName' => $review->getReviewSubject()->getName(),]);
```
Comment (GSadee) [inline src/ProductReview/Mailer/ConfirmationMailer.php]: Remove `,`

File: src/ProductReview/Mailer/ConfirmationMailerInterface.php [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+declare(strict_types=1);
+
+namespace App\ProductReview\Mailer;
+
+use Sylius\Component\Review\Model\ReviewInterface;
+
+interface ConfirmationMailerInterface
+{
+    public function sendEmail(ReviewInterface $reviewer): void;
```
Comment (GSadee) [inline src/ProductReview/Mailer/ConfirmationMailerInterface.php]: `$reviewer` -> `$review`

File: templates/Emails/reviewConfirmationNotification.html.twig [hunk 8]
```diff
@@ -0,0 +1,7 @@
+{% block subject %}
+    Your review of {{ productName }} has been added.
+{% endblock %}
+
+{% block body %}
+Your review has been added Yeah !
```
Comment (GSadee) [inline templates/Emails/reviewConfirmationNotification.html.twig]: Wrong indentation

Comment (GSadee) [inline templates/Emails/reviewConfirmationNotification.html.twig]: <del>Yeah</del>?

File: config/packages/test/behat.xml [hunk 9]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+    <services>
+        <defaults public="true" />
+
+        <service id="app.behat.context.ui.email" class="App\Tests\Behat\Context\Ui\EmailContext">
+            <argument type="service" id="sylius.behat.email_checker" />
+        </service>
+
```
Comment (Zales0123) [inline config/packages/test/behat.xml]: Redundant blank line

File: config/packages/sylius_mailer.yaml [hunk 10]
```diff
@@ -0,0 +1,8 @@
+sylius_mailer:
+    sender:
+        name: Review confirmation
+        address: example@example.com
```
Comment (Zales0123) [inline config/packages/sylius_mailer.yaml]: Maybe something more realistic? Like `admin@example.com`? :)

File: src/ProductReview/Mailer/ConfirmationMailer.php [hunk 11]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
```
Comment (Zales0123) [inline src/ProductReview/Mailer/ConfirmationMailer.php]: We should not add license blocks in customization recommendations

File: src/ProductReview/Mailer/ConfirmationMailer.php [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+    public function __construct(SenderInterface $emailSender)
+    {
+        $this->emailSender = $emailSender;
+    }
+
+    public function sendEmail(ReviewInterface $review): void
+    {
+        $this->emailSender->send('review_confirmation',
+            [$review->getAuthor()->getEmail()],
+            ['productName' => $review->getReviewSubject()->getName()]);
```
Comment (Zales0123) [inline src/ProductReview/Mailer/ConfirmationMailer.php]: Move colon to the next line

File: tests/Behat/Context/Ui/EmailContext.php [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        $this->emailChecker = $emailChecker;
+    }
+
+    /**
+     * @Then an email with review confirmation should be sent to :emailAddress
+     */
+    public function anEmailWithReviewConfirmationShouldBeSentTo(string $emailAddress): void
+    {
+        Assert::true($this->emailChecker->hasMessageTo('Your review has been added Yeah !', $emailAddress));
```
Comment (Zales0123) [inline tests/Behat/Context/Ui/EmailContext.php]: The message here is not the same as in the email template 💃

File: config/packages/sylius_mailer.yaml [hunk 14]
```diff
@@ -0,0 +1,8 @@
...
+    emails:
+        review_confirmation:
+            subject: A new review has been added
```
Comment (Zales0123) [inline config/packages/sylius_mailer.yaml]: `Your review has just been accepted!`?

File: templates/Emails/reviewConfirmation.html.twig [hunk 15]
```diff
@@ -0,0 +1,7 @@
+{% block subject %}
+    Your review of {{ productName }} has been added.
+{% endblock %}
+
+{% block body %}
+    Your review has been added.
```
Comment (Zales0123) [inline templates/Emails/reviewConfirmation.html.twig]: `added` -> `accepted`? Here and above

**Pull request closed without merging**